### PR TITLE
ENH: natively support Graph in moran module

### PR DESCRIPTION
**Pull request by martinfleis**
Adding native support of Graph as an input for `w` in classes within the `moran.py` module. Wasn't very hard and no test failed. I call that a success.

Parametrising tests is a bit pain but doable.

This will fail until https://github.com/pysal/libpysal/pull/742 is merged. Shall we try to capture libpysal version and raise if s0... are not available? I think it affects only `Moran` and its subclasses here.

I have also taken the libery of updating `test_moran` to follow pytest convention, primarily to make `parametrize` work.

I think that the same treatment could be applied to all remaining modules here and we may even avoid `Graph.to_W()` path here.

### CODE REVIEW

**Review by jGaboardi — APPROVED** [review]

**Review by sjsrey — APPROVED** [review]

**Comment by martinfleis** [comment]
I've marked this as a draft until the relevant features will land in lib.

File: esda/moran.py
Comment (sjsrey) [inline esda/moran.py:234]: This seems like a less elegant approach then having s0, s1, s2 be public on the graph? The latter would avoid the conditional (I think)?

Comment (martinfleis) [reply esda/moran.py:234]: It is byt of we want to deprecate public properties than it is a way forward. As I said in the libpysal PR I don't mind which way we go.

**Comment by jGaboardi** [comment]
@martinfleis let's update with `main` and rerun tests

**Comment by martinfleis** [comment]
Dev is green 🎉  What do you want me to do with the rest? Wait until lib 4.12 and pin?

**Review by knaaptime — APPROVED** [review]

**Comment by jGaboardi** [comment]
* CI is all green - `libpysal` is pinned - we have approvals
* I'll merge tomorrow in the AM unless someone else already has or there are detractors

**Pull request merged**
